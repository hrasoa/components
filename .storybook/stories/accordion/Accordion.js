// @flow
import * as React from 'react';
import {
  AccordionProvider,
  Accordion,
  AccordionHeader,
  AccordionPanel,
} from '../../../packages/accordion/src';

type Props = {
  renderHeader?: (i: number) => string | React.Node,
  className?: string,
  allowTogle?: boolean,
  allowMultiple?: boolean,
  expandedIds?: Array<string>,
};

const getItems = (expandedIds, renderHeader) => {
  let items = [];
  for (let i = 0; i <= 2; i += 1) {
    items = [
      ...items,
      <AccordionHeader
        key={`header-${i}`}
        id={`header-${i}`}
        controls={`panel-${i}`}
      >
        {renderHeader && renderHeader(i)}
      </AccordionHeader>,
      <AccordionPanel
        key={`panel-${i}`}
        id={`panel-${i}`}
        isExpanded={expandedIds && expandedIds.indexOf(i) >= 0}
      >
        <ul>
          <li><label htmlFor={`input-${i}-0`}>Label <input id={`input-${i}-0`} /></label></li>
          <li><label htmlFor={`input-${i}-1`}>Label <input id={`input-${i}-1`} /></label></li>
        </ul>
      </AccordionPanel>,
    ];
  }
  return items;
};

const AccordionTest = ({ expandedIds, renderHeader, ...props }: Props) => (
  <AccordionProvider {...props}>
    <Accordion>
      {getItems(expandedIds, renderHeader)}
    </Accordion>
  </AccordionProvider>
);

AccordionTest.defaultProps = {
  className: null,
  allowMultiple: false,
  allowTogle: false,
  expandedIds: null,
  renderHeader: i => `Header ${i + 1}`,
};

export default AccordionTest;
