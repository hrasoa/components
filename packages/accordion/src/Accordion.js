// @flow
import * as React from 'react';

type Props = {
  children: string | React.Node,
  handleKeyNavigation: (e: SyntheticKeyboardEventElement<HTMLElement>) => void,
  className?: ?string,
};

/** Accordion descripiion */
const Accordion = (props: Props) => (
  <dl
    className={props.className}
    role="presentation"
    onKeyDown={props.handleKeyNavigation}
  >
    {props.children}
  </dl>
);

Accordion.defaultProps = {
  className: null,
};

export default Accordion;
