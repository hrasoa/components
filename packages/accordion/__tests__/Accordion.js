// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Accordion from '../src/Accordion';

const mockKeyNavigation = jest.fn();

beforeEach(() => {
  mockKeyNavigation.mockClear();
});

describe('Accordion', () => {
  describe('Snapshot testing', () => {
    test('minimal', () => {
      const component = renderer.create(
        <Accordion
          handleKeyNavigation={mockKeyNavigation}
        >
          <span>accordion items</span>
        </Accordion>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    test('with more props', () => {
      const component = renderer.create(
        <Accordion
          handleKeyNavigation={mockKeyNavigation}
          className="accordion"
          style={{ color: 'inherit' }}
        >
          <span>accordion items</span>
        </Accordion>,
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('DOM Testing', () => {
    test('should call handleKeyNavigation onKeyDown', () => {
      const wrapper = shallow(
        <Accordion
          handleKeyNavigation={mockKeyNavigation}
        >
          <span>accordion items</span>
        </Accordion>,
      );
      wrapper.simulate('keydown');
      expect(mockKeyNavigation).toHaveBeenCalledTimes(1);
    });
  });
});
