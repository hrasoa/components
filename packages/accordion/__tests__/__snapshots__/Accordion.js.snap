// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Snapshot testing minimal 1`] = `
<dl
  className={null}
  onKeyDown={[MockFunction]}
  role="presentation"
>
  <span>
    accordion items
  </span>
</dl>
`;

exports[`Accordion Snapshot testing touched state 1`] = `
<dl
  className="is-touched"
  onKeyDown={[MockFunction]}
  role="presentation"
>
  <span>
    accordion items
  </span>
</dl>
`;

exports[`Accordion Snapshot testing with more props 1`] = `
<dl
  className="accordion"
  onKeyDown={[MockFunction]}
  role="presentation"
  style={
    Object {
      "color": "inherit",
    }
  }
>
  <span>
    accordion items
  </span>
</dl>
`;
